runtime-modules/bounty/src/lib.rs

```diff
@@ -1587,52 +1940,59 @@ impl<T: Config> Module<T> {
         Ok(bounty)
     }
 
+    fn ensure_bounty_contribution_exists(
+        bounty_id: &T::BountyId,
+        funder: &BountyActor<MemberId<T>>,
+    ) -> Result<Contribution<T>, DispatchError> {
+        ensure!(
+            <BountyContributions<T>>::contains_key(&bounty_id, &funder),
+            Error::<T>::NoBountyContributionFound,
+        );
+
+        let funding = <BountyContributions<T>>::get(&bounty_id, &funder);
+
+        Ok(funding)
+    }
     // Calculate cherry fraction to reward member for an unsuccessful bounty.
     // Cherry fraction = cherry * (member funding / total funding).
     fn get_cherry_fraction_for_member(
         bounty: &Bounty<T>,
         funding_amount: BalanceOf<T>,
     ) -> BalanceOf<T> {
-        let funding_share =
-            Perbill::from_rational_approximation(funding_amount, bounty.total_funding);
+        let funding_share = Perbill::from_rational(funding_amount, bounty.total_funding);
 
         // cherry share
         funding_share * bounty.creation_params.cherry
     }
 
-    // Remove bounty and all related info from the storage.
-    fn remove_bounty(bounty_id: &T::BountyId) {
-        <Bounties<T>>::remove(bounty_id);
-        <BountyContributions<T>>::remove_prefix(bounty_id);
+    /// Remove bounty and all related info from the storage.
+    fn remove_bounty(
+        bounty_id: &T::BountyId,
+        bounty: &Bounty<T>,
+        bounty_creator_manager: &BountyActorManager<T>,
+    ) {
+        bounty_creator_manager
+            .transfer_funds_from_bounty_account(*bounty_id, T::CreatorStateBloatBondAmount::get());
 
-        // Slash remaining funds.
-        let bounty_account_id = Self::bounty_account_id(*bounty_id);
-        let all = balances::Pallet::<T>::usable_balance(&bounty_account_id);
-        if all != Zero::zero() {
-            let _ = balances::Pallet::<T>::slash(&bounty_account_id, all);
-        }
+        Self::deposit_event(RawEvent::CreatorStateBloatBondWithdrawn(
+            *bounty_id,
+            bounty.creation_params.creator.clone(),
+            T::CreatorStateBloatBondAmount::get(),
+        ));
+
+        <Bounties<T>>::remove(bounty_id);
+        <BountyContributions<T>>::remove_prefix(bounty_id, None);
```

Comment:
We will have to implement another map to map a bountyId to the contributions count

Reply:
@bedeho

Reply:
I guess this is not a problem, since this is called in `remove_bounty` and this function must be called after 
`has_no_contributions_and_no_work_entries`, so there should be no contributions at this stage.
In fact, remove_prefix is not needed, I tested and after removing all contributions the keys were also removed